Feature/snake game

mercurrie opened the pull request:
included main.py, scoreboard.py, snake.py, and food.py files

mercurrie committed:
snake body

mercurrie committed:
snake move

create main.py and snake.py to perform basic movements of snake

mercurrie committed:
food

mercurrie committed:
feature

created class for food, and scoreboard.
complete program

michaelhousman commented on food.py:
```diff
@@ -0,0 +1,20 @@
+class Food(Turtle):
+
+    def __init__(self):
+        super().__init__()
```
Looks fine but my biggest critique is that I like to stash constants at the top of the page so that they're not buried in functions.

mercurrie replied:
added constants to json file

michaelhousman commented on scoreboard.py:
```diff
@@ -0,0 +1,30 @@
+from turtle import Turtle
+ALIGNMENT = "center"
```
This is better

michaelhousman commented on snake.py:
```diff
@@ -0,0 +1,61 @@
+from turtle import Turtle
+STARTING_POSITIONS = [(0, 0), (-20, 0), (-40, 0)]
```
When you stash at the top, you can use a mapping and then all you have to update is that.  That also allows you to stash in a JSON (instead of `.py` code), which means that you won't have to run a deploy if you want to make changes.  That's how you can make the code super flexible.

mercurrie replied:
imported json file for constants

michaelhousman reviewed (changes requested):
Have one big change to request.

mercurrie committed:
json

included json file for constants

michaelhousman commented on scoreboard.py:
```diff
@@ -0,0 +1,30 @@
+    def __init__(self):
+        super().__init__()
+        self.score = 0
+        self.color("white")
```
Still some constants here.

michaelhousman commented on snake.py:
```diff
@@ -0,0 +1,58 @@
+    def add_segment(self, position: tuple):
+        snake_fragment = Turtle(shape="square")
+        snake_fragment.penup()
+        snake_fragment.color("white")
```
And here. But I think you get it.

mercurrie replied:
flushed out the rest of the constants, and  I included a json file to read and write the high score to

michaelhousman reviewed (approved):
Still some constants present but I think you get it.

mercurrie committed:
constants

turned hard coded values into constants within json file

mercurrie committed:
constants, and high score

included json file containing all constant variables.

included json file to read and write high score
